api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -80,14 +80,9 @@ public User createUser(User user, String password) throws APIException {
 		
 		checkPrivileges(user);
 		
-		// if a password wasn't supplied, throw an error
-		if (password == null || password.length() < 1) {
```

Comment:
This gets implicitly tested by the RequiredDataAdvice.
Even though I think it's less transparent using AOP for null/emptiness checks, leaving this code (that doesn't have an impact) in there would lead to confusion, I think.